h/queue.py

```diff
@@ -26,18 +33,22 @@ def get_reader(settings, topic, channel):
     The caller is responsible for adding appropriate `on_message` hooks and
     starting the reader.
     """
+    req.sentry.extra_context(extra_context_data(req, topic))
+    settings = req.registry.settings
     addrs = aslist(settings.get('nsq.reader.addresses', 'localhost:4150'))
     topic = resolve_topic(topic, settings=settings)
     reader = gnsq.Reader(topic, channel, nsqd_tcp_addresses=addrs)
     return reader
 
 
-def get_writer(settings):
+def get_writer(req):
     """
     Get a :py:class:`gnsq.Nsqd` instance configured to connect to the nsqd
     writer address configured in settings. The writer communicates over the
     nsq HTTP API and does not hold a connection open to the nsq instance.
     """
+    req.sentry.extra_context(extra_context_data(req))
```

Comment:
We don't really have any context in this scenario, because a writer is ephemeral and can be used to write to any topic.